cupy/core/reduction.pxi

```diff
@@ -32,15 +32,16 @@ extern "C" __global__ void ${name}(${params}) {
   unsigned int _tid = threadIdx.x;
 
   int _J_offset = _tid >> __popc(_block_stride - 1);  // _tid / _block_stride
-  int _j_offset = _J_offset * _out_ind.size();
+  long long _j_offset = (long long)_J_offset * _out_ind.size();
```

Comment:
`ptrdiff_t` might be better than `long long` from the point of view of consistency.
https://github.com/kmaehashi/cupy/blob/fix-memory-corruption/cupy/core/include/cupy/carray.cuh#L167

Reply:
Agree with that, fixed.